BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/service/PendingSubmissionUtil.java

```diff
@@ -0,0 +1,128 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@Component
+@Slf4j
+@AllArgsConstructor
+public class PendingSubmissionUtil {
+
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    private static final String ACCNO_PREFIX = "TMP_";
+
+    private final ObjectMapper objectMapper;
+
+    public String asString(PendingSubmissionDto dto) {
+        return objectMapper.valueToTree(dto).toString();
+    }
+
+    public Optional<PendingSubmissionDto> parse(String data) {
+        try {
+            return Optional.of(objectMapper.readValue(data, PendingSubmissionDto.class));
+        } catch (IOException e) {
+            log.error("error while parsing pending submission", e);
+        }
+        return Optional.empty();
+    }
+
+    public PendingSubmissionListItemDto convert(PendingSubmissionDto pendingSubmission) {
+        JsonNode node = pendingSubmission.getData();
+        return PendingSubmissionListItemDto.builder()
+                .accno(pendingSubmission.getAccno())
+                .mtime(pendingSubmission.getModificationTimeInSeconds())
+                .rtime(getReleaseTimeInSeconds(node).orElse(null))
+                .title(getTitle(node).orElse(""))
+                .build();
+    }
+
+    public Optional<PendingSubmissionDto> create(String pageTab) {
+        try {
+            JsonNode node = objectMapper.readTree(pageTab);
+            PendingSubmissionDto submission = new PendingSubmissionDto();
+            submission.setAccno(getAccno(node).orElse(newAccno()));
+            submission.setData(node);
+            submission.setChanged(System.currentTimeMillis());
+            return Optional.of(submission);
+
+        } catch (IOException e) {
+            log.error("error while creating pending submission", e);
+        }
+        return Optional.empty();
+    }
+
+    static private String newAccno() {
+        return ACCNO_PREFIX + System.currentTimeMillis();
+    }
+
+    static private Optional<String> getAccno(JsonNode jsonNode) {
+        return Optional.ofNullable(jsonNode.get("accno")).map(JsonNode::asText).map(String::trim).filter(s -> !s.isEmpty());
+    }
+
+    static private Optional<String> getTitle(JsonNode jsonNode) {
+        Optional<JsonNode> attrValueNode = attributes(jsonNode)
+                .stream()
+                .filter(attrNameFilter("title"))
+                .findFirst();
+        return attrValueNode.map(v -> attrValue(v, ""));
+    }
+
+    static private Optional<Long> getReleaseTimeInSeconds(JsonNode node) {
+        Optional<JsonNode> attrValueNode = attributes(node)
+                .stream()
+                .filter(attrNameFilter("releaseDate"))
+                .findFirst();
+
+        return attrValueNode.map(v -> numberOfSeconds(attrValue(v, "")));
+    }
+
+    static private List<JsonNode> attributes(JsonNode node) {
```

Comment:
all this methods do not be to be static.

Reply:
True. But Its good to know that something can be static :-)  Done